docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/NormalizeBinning.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+using static Microsoft.ML.Transforms.NormalizingTransformer;
+
+namespace Samples.Dynamic
+{
+    public class NormalizeBinning
+    {
+        public static void Example()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var mlContext = new MLContext();
+            var samples = new List<DataPoint>()
+            {
+                new DataPoint(){ Features = new float[4] { 8, 1, 3, 0} },
+                new DataPoint(){ Features = new float[4] { 6, 2, 2, 0} },
+                new DataPoint(){ Features = new float[4] { 4, 0, 1, 0} },
+                new DataPoint(){ Features = new float[4] { 2,-1,-1, 1} }
+            };
+            // Convert training data to IDataView, the general data type used in ML.NET.
+            var data = mlContext.Data.LoadFromEnumerable(samples);
+            // NormalizeBinning normalizes the data by constructing equidensity bins and produce output based on 
+            // to which bin original value belong.
+            var normalize = mlContext.Transforms.NormalizeBinning("Features", maximumBinCount: 4, fixZero: false);
+
+            // NormalizeBinning normalizes the data by constructing equidensity bins and produce output based on 
+            // to which bin original value belong but make sure zero values would remain zero after normalization.
+            // Helps preserve sparsity.
+            var normalizeFixZero = mlContext.Transforms.NormalizeBinning("Features", maximumBinCount: 4, fixZero: true);
+
+            // Now we can transform the data and look at the output to confirm the behavior of the estimator.
+            // This operation doesn't actually evaluate data until we read the data below.
+            var normalizeTransform = normalize.Fit(data);
+            var transformedData = normalizeTransform.Transform(data);
+            var normalizeFixZeroTransform = normalizeFixZero.Fit(data);
+            var fixZeroData = normalizeFixZeroTransform.Transform(data);
+            var column = transformedData.GetColumn<float[]>("Features").ToArray();
+            foreach (var row in column)
+                Console.WriteLine(string.Join(", ", row.Select(x => x.ToString("f4"))));
+            // Expected output:
+            //  1.0000, 0.6667, 1.0000, 0.0000
+            //  0.6667, 1.0000, 0.6667, 0.0000
+            //  0.3333, 0.3333, 0.3333, 0.0000
```

Comment:
>0.3333 [](start = 24, length = 6)

I'd emphasize this value.

Reply:
I don't get your comment.

---
In reply to: [273786583](https://github.com/dotnet/machinelearning/pull/3244#discussion_r273786583) [](ancestors = 273786583)